recipes/flatbuffers/all/conanfile.py

```diff
@@ -57,86 +48,82 @@ def configure(self):
             del self.options.fPIC
         if self.options.header_only:
             del self.options.shared
-        # deprecated options
-        for deprecated_option in ["flatc", "flatbuffers", "options_from_context"]:
-            if self.options.get_safe(deprecated_option) != "deprecated":
-                self.output.warn("{} option is deprecated, do not use".format(deprecated_option))
+
+    def package_id(self):
+        if self.options.header_only and not self._has_flatc:
+            self.info.clear()
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
+            check_min_cppstd(self, 11)
 
-    def package_id(self):
-        if self.options.header_only and not self._has_flatc:
-            self.info.header_only()
-        # deprecated options
-        del self.info.options.flatc
-        del self.info.options.flatbuffers
-        del self.info.options.options_from_context
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["FLATBUFFERS_BUILD_TESTS"] = False
+        tc.variables["FLATBUFFERS_INSTALL"] = True
+        tc.variables["FLATBUFFERS_BUILD_FLATLIB"] = not self.options.header_only and not self.options.shared
+        tc.variables["FLATBUFFERS_BUILD_FLATC"] = self._has_flatc
+        tc.variables["FLATBUFFERS_STATIC_FLATC"] = False
+        tc.variables["FLATBUFFERS_BUILD_FLATHASH"] = False
+        tc.variables["FLATBUFFERS_BUILD_SHAREDLIB"] = not self.options.header_only and self.options.shared
+        # Honor conan profile
+        tc.variables["FLATBUFFERS_LIBCXX_WITH_CLANG"] = False
+        # Mimic upstream CMake/Version.cmake removed in _patch_sources()
+        version = Version(self.version)
+        tc.variables["VERSION_MAJOR"] = version.major
+        tc.variables["VERSION_MINOR"] = version.minor
+        tc.variables["VERSION_PATCH"] = version.patch
+        # For msvc shared
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+        # Relocatable shared libs on Macos
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
```

Comment:
What do you think the advice should be for deciding on using `cache variables`? The [conan docs](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html?highlight=cache_variable#cache-variables) are not the most intuitive

I want to say 

> anything that needs to be set before the call to `project` or `cmake_min_version`

 but that's not super obvious -- so my current train of thought is

> whenever you use set a `CMAKE_` prefixed var

I havent seen any downside yet but Looking for a second opinion 🙏 

```suggestion
        tc.cache_variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
        # Relocatable shared libs on Macos
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
```

Reply:
I've explained some cases in a conan issue, but can't remember which one.

Reply:
Thank you for the link!